src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,120 @@
+        directlyOkLogin = new User("abc123", "pass1test", 19);
+        directlyOkPassword = new User("login_test", "abc321", 20);
+        nullUser = null;
+        userLoginNull = new User(null, "pass4test", 45);
+        userPasswordNull = new User("test_login1", null, 45);
+        userAgeNull = new User("test_login2", "pass4test", null);
+        emptyLogin = invalidLogin = new User("", "pass4test", 45);
+        invalidLogin = new User("abc", "pass4test", 45);
+        barelyOkLogin = new User("abc12", "pass4test", 45);
+        emptyPassword = new User("test_login", "", 45);
+        invalidPassword = new User("test_login3", "abc", 45);
+        barelyOkPassword = new User("test_login3", "wrong", 45);
+        invalidAge = new User("test_login", "pass4test", -5);
+        tooYoungAge = new User("test_login", "pass4test", 16);
+        directlyOkAge = new User("test_age", "pass4age", DEFAULT_AGE);
+    }
+
+    @Test
+    void checkNull() {
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(nullUser),
+                "Expected: invalidDataException when User is null \n");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userLoginNull),
+                "Expected: invalidDataException when User login is null \n");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userPasswordNull),
+                "Expected: invalidDataException when User password is null \n");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(userAgeNull),
+                "Expected: invalidDataException when User age is null \n");
+    }
+
+    @Test
+    void checkInvalidLogin() {
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(emptyLogin),
+                "User login can't be empty");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(invalidLogin),
+                "User login must have at least 6 characters");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(barelyOkLogin),
+                "User login must have at least 6 characters");
+    }
+
+    @Test
+    void checkInvalidPassword() {
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(emptyPassword),
+                "User password can't be empty");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(invalidPassword),
+                "User password must have at least 6 characters");
+        assertThrows(InvalidDataException.class,
+                () -> registrationService.register(barelyOkPassword),
+                "User password must have at least 6 characters");
+    }
+
+    @Test
+    void checkInvalidAge() {
```

Comment:
adhere to the suggested naming convention
![image](https://github.com/mate-academy/jv-registration-validation/assets/49767049/9995d1e8-eb92-490b-b849-f819dcbb6a2f)